gridcapa-task-manager-app/src/test/java/com/farao_community/farao/gridcapa/task_manager/app/TaskUpdateNotifierTest.java

```diff
@@ -0,0 +1,128 @@
+import static org.mockito.Mockito.anyBoolean;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.contains;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class TaskUpdateNotifierTest {
+
+    @Mock
+    private StreamBridge streamBridge;
+
+    @Mock
+    private TaskDtoBuilderService taskDtoBuilderService;
+
+    @Mock
+    private SimpMessagingTemplate stompBridge;
+
+    @Mock
+    private WebsocketConfig websocketConfig;
+
+    @InjectMocks
+    private TaskUpdateNotifier taskUpdateNotifier;
+
+    private Task task;
+    private TaskDto taskDto;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        task = mock(Task.class);
+        taskDto = mock(TaskDto.class);
+        when(taskDtoBuilderService.createDtoFromEntityWithoutProcessEvents(any(Task.class))).thenReturn(taskDto);
+        when(task.getTimestamp()).thenReturn(OffsetDateTime.now());
+        when(websocketConfig.getNotify()).thenReturn("/topic");
+    }
+
+    @Test
+    void testNotifyWithStatusUpdate() {
+        // Given
+        final boolean withStatusUpdate = true;
+        final boolean withEventsUpdate = false;
+        final boolean withNewInput = false;
+
+        // When
+        taskUpdateNotifier.notify(task, withStatusUpdate, withEventsUpdate, withNewInput);
+
+        // Then
+        verify(streamBridge, times(1)).send("task-status-updated", taskDto);
+        verify(stompBridge, times(2)).convertAndSend(anyString(), eq(taskDto));
+        verify(stompBridge, never()).convertAndSend(contains("/events"), anyBoolean());
+    }
+
+    @Test
+    void testNotifyWithNewInput() {
+        // Given
+        final boolean withStatusUpdate = false;
+        final boolean withEventsUpdate = false;
+        final boolean withNewInput = true;
+
+        // When
+        taskUpdateNotifier.notify(task, withStatusUpdate, withEventsUpdate, withNewInput);
+
+        // Then
+        verify(streamBridge, times(1)).send("task-input-updated", taskDto);
+        verify(stompBridge, times(2)).convertAndSend(anyString(), eq(taskDto));
+        verify(stompBridge, never()).convertAndSend(contains("/events"), anyBoolean());
+    }
+
+    @Test
+    void testNotifyWithEventsUpdate() {
+        // Given
+        final boolean withStatusUpdate = false;
+        final boolean withEventsUpdate = true;
+        final boolean withNewInput = false;
+
+        // When
+        taskUpdateNotifier.notify(task, withStatusUpdate, withEventsUpdate, withNewInput);
+
+        // Then
+        verify(streamBridge, never()).send(anyString(), any());
+        verify(stompBridge, times(2)).convertAndSend(anyString(), eq(taskDto));
+        verify(stompBridge, times(1)).convertAndSend(contains("/events"), eq(true));
+    }
+
+    @Test
+    void testNotifyWithoutUpdates() {
```

Comment:
I don't know if that would be relevant, but you could add a test case with StatusUpdate, EventsUpdate and NewInput all set to true. Such a case can occur in TaskUpdateNotifier and testing it would be a way to ensure that different types of updates are not exclusive.